Changing precision alters which sites are flagged for full inference?

Very weirdly, if I change the default precision value to 13 here:

https://github.com/tskit-dev/tsinfer/blob/7c23e61425254674b1caafa4c7419851d647141b/tsinfer/inference.py#L896

Then this test fails:

```
pytest tests/test_inference.py::TestMetadataRoundTrip::test_site_metadata
...
>               assert value == tsinfer.INFERENCE_FULL
E               AssertionError: assert 'parsimony' == 'full'
E                 - full
E                 + parsimony

tests/test_inference.py:819: AssertionError
============================================================== short test summary info ===============================================================
FAILED tests/test_inference.py::TestMetadataRoundTrip::test_site_metadata - AssertionError: assert 'parsimony' == 'full'
```

I have no idea why that should be

--- a/tests/test_inference.py
+++ b/tests/test_inference.py
@@ -791,13 +791,13 @@ class TestMetadataRoundTrip:
             assert all_metadata[variant.site.id] == variant.site.metadata
 
         output_ts = tsinfer.infer(sample_data)
-        samples = set(output_ts.samples())
-        for site in output_ts.sites():
+        for variant in output_ts.variants():
+            site = variant.site
             decoded_metadata = json.loads(site.metadata)
             assert "inference_type" in decoded_metadata
             value = decoded_metadata.pop("inference_type")
             # Only singletons should be parsimony sites in this simple case
-            if len(site.mutations) == 1 and site.mutations[0].node in samples:
+            if np.sum(variant.genotypes > 0) == 1:
                 assert value == tsinfer.INFERENCE_PARSIMONY
             else:
                 assert value == tsinfer.INFERENCE_FULL

